kytos/topology
# Fix the interface metadata persistence

This commit made a small refactor in the `save_metadata_on_store` method and the `_restore_status`, making the interface metadata persistent. Solving a bug to save the interface metadata in the NApp storehouse.

Main change:

- Added call to the `update_instance` method in `restore_status_method`, this allows update the interface instance with saved metadata.
-  Refactoring in the `save_metadata_on_store` method.

**How test these changes:**

- Run kytos (`kytosd -f`).
- Start a network topology (e.g , `mn - linear top, 3 --controller = remote, ip = 127.0.0.1, port = 6653`)
- Add metadata to the interface:
   - `POST http://0.0.0.0:8181/api/kytos/topology/v3/interfaces/<interface_id>/metadata`
      - Payload `{ 'test':'metadata'}`
- Close the kytos (`exit`)
- Start kytos again.
- Restore changes (http://127.0.0.1:8181/api/kytos/topology/v3/restore), this will be automatic soon (#123 )
- Check the interface metadata.

Fix: #129
